### PR TITLE
Require Maven 3.2.5+

### PR DESCRIPTION
**Pull request by slawekjaranowski**
- use Maven 3.2.5 api
- migrate plexus annotation to JSR330

---

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MSHARED) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes. Also be sure having selected the correct component.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MSHARED-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MSHARED-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -59,51 +59,50 @@
...
   <properties>
-    <mavenVersion>2.0.5</mavenVersion>
-    <surefire.version>2.22.2</surefire.version>
+    <mavenVersion>3.2.5</mavenVersion>
     <javaVersion>8</javaVersion>
     <project.build.outputTimestamp>2022-01-24T19:57:26Z</project.build.outputTimestamp>
   </properties>
 
   <dependencies>
+
```
Comment (michael-o) [inline pom.xml:68]: They don't end with an empty line as well; inconsistent

File: src/main/java/org/apache/maven/shared/dependency/analyzer/DefaultProjectDependencyAnalyzer.java [hunk 2]
```diff
@@ -94,7 +97,7 @@ public ProjectDependencyAnalysis analyze( MavenProject project )
             Set<Artifact> testArtifactsWithNonTestScope = getTestArtifactsWithNonTestScope( testOnlyArtifacts );
 
             return new ProjectDependencyAnalysis( usedDeclaredArtifacts, usedUndeclaredArtifactsWithClasses,
-                                                  unusedDeclaredArtifacts, testArtifactsWithNonTestScope );
+                unusedDeclaredArtifacts, testArtifactsWithNonTestScope );
```
Comment (michael-o) [inline src/main/java/org/apache/maven/shared/dependency/analyzer/DefaultProjectDependencyAnalyzer.java:100]: Contradicts our style

File: src/main/java/org/apache/maven/shared/dependency/analyzer/DefaultProjectDependencyAnalyzer.java [hunk 3]
```diff
@@ -263,7 +266,7 @@ else if ( file != null && file.isDirectory() )
     }
 
     private Map<Artifact, Set<String>> buildUsedArtifacts( Map<Artifact, Set<String>> artifactClassMap,
-                                              Set<String> dependencyClasses )
+                                                           Set<String> dependencyClasses )
```
Comment (michael-o) [inline src/main/java/org/apache/maven/shared/dependency/analyzer/DefaultProjectDependencyAnalyzer.java:269]: Same here

File: src/main/java/org/apache/maven/shared/dependency/analyzer/DefaultProjectDependencyAnalyzer.java (hunk 2 above)
Comment (slawekjaranowski) [reply src/main/java/org/apache/maven/shared/dependency/analyzer/DefaultProjectDependencyAnalyzer.java:100]: multiline arguments should be aligned for method call - ok

There are no examles / rules ... https://maven.apache.org/developers/conventions/code.html

File: src/main/java/org/apache/maven/shared/dependency/analyzer/DefaultProjectDependencyAnalyzer.java (hunk 3 above)
Comment (michael-o) [reply src/main/java/org/apache/maven/shared/dependency/analyzer/DefaultProjectDependencyAnalyzer.java:269]: I remember that we do this everywhere although not being consistent

Comment (slawekjaranowski) [reply src/main/java/org/apache/maven/shared/dependency/analyzer/DefaultProjectDependencyAnalyzer.java:269]: so 
 - arguments for multiline method declaration - should be aligned
 - arguments for multiline method call - should be not aligned

corect?

Comment (slawekjaranowski) [reply src/main/java/org/apache/maven/shared/dependency/analyzer/DefaultProjectDependencyAnalyzer.java:269]: so 
 - arguments for multiline method declaration - should be aligned
 - arguments for multiline method call - should be not aligned

corect?

**Review by slachiewicz — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (slawekjaranowski) [reply pom.xml:68]: ok

File: src/main/java/org/apache/maven/shared/dependency/analyzer/DefaultProjectDependencyAnalyzer.java (hunk 3 above)
Comment (slawekjaranowski) [reply src/main/java/org/apache/maven/shared/dependency/analyzer/DefaultProjectDependencyAnalyzer.java:269]: @michael-o - what  formatting should be done ... I think that my proposition is correct

File: src/main/java/org/apache/maven/shared/dependency/analyzer/DefaultProjectDependencyAnalyzer.java (hunk 2 above)
Comment (michael-o) [reply src/main/java/org/apache/maven/shared/dependency/analyzer/DefaultProjectDependencyAnalyzer.java:100]: It's incomplete. What I have seen in Core and other components is to align on the same column

**Review by michael-o — APPROVED** [review]
Fine except commented formatting

File: src/main/java/org/apache/maven/shared/dependency/analyzer/asm/DependencyClassFileVisitor.java [hunk 4]
```diff
@@ -56,7 +56,7 @@ public void visitClass( String className, InputStream in )
     {
         try
         {
-            byte[] byteCode = IOUtils.toByteArray( in );
+            byte[] byteCode = IOUtil.toByteArray( in );
```
Comment (michael-o) [inline src/main/java/org/apache/maven/shared/dependency/analyzer/asm/DependencyClassFileVisitor.java:59]: This can also be done with NIO2. No need for P U

Comment (slawekjaranowski) [reply src/main/java/org/apache/maven/shared/dependency/analyzer/asm/DependencyClassFileVisitor.java:59]: It is InputStream ... 
plexus utils is used anyway in other place. This was easy replace, allowing to reduce one dependency from compile scope

**Pull request merged**